pkg/yang/types_builtin_test.go

```diff
@@ -0,0 +1,634 @@
+// Copyright 2015 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yang
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/openconfig/gnmi/errdiff"
+)
+
+const (
+	useMin = -999
+	useMax = 999
+)
+
+// R is a test helper for creating an int-based YRange.
+func R(a, b int64) YRange {
+	n1 := FromInt(a)
+	n2 := FromInt(b)
+	if a == useMin {
+		n1 = minNumber
+	}
+	if b == useMax {
+		n2 = maxNumber
+	}
+	return YRange{n1, n2}
+}
+
+// Rf is a test helper for creating a float-based YRange.
+func Rf(a, b int64, fracDig uint8) YRange {
+	n1 := Number{Value: uint64(a), FractionDigits: fracDig}
+	n2 := Number{Value: uint64(b), FractionDigits: fracDig}
+	if a < 0 {
+		n1.Value = uint64(-a)
+		n1.Kind = Negative
+	}
+	if b < 0 {
+		n2.Value = uint64(-b)
+		n2.Kind = Negative
+	}
+	if a == useMin {
+		n1 = minNumber
+	}
+	if b == useMax {
+		n2 = maxNumber
+	}
+	return YRange{n1, n2}
+}
+
+func TestRangeEqual(t *testing.T) {
+	for x, tt := range []struct {
+		r1, r2 YangRange
```

Comment:
Whilst we're tweaking these, should we name them in a way that would be more typical? i.e., `inTestRange` and `inContainedRange`? We could also make `ok` -> `want` per usual conventions.